cmake/developer_package/plugins/plugins.cmake

```diff
@@ -146,39 +147,39 @@ function(ie_add_plugin)
         endif()
     endif()
 
-    # check that plugin with such name is not registered
-
-    foreach(plugin_entry IN LISTS PLUGIN_FILES)
-        string(REPLACE ":" ";" plugin_entry "${plugin_entry}")
-        list(GET plugin_entry -1 library_name)
-        list(GET plugin_entry 0 plugin_name)
-        if(plugin_name STREQUAL "${IE_PLUGIN_DEVICE_NAME}" AND
-           NOT library_name STREQUAL ${IE_PLUGIN_NAME})
-            message(FATAL_ERROR "${IE_PLUGIN_NAME} and ${library_name} are both registered as ${plugin_name}")
-        endif()
-    endforeach()
+    if(NOT IE_PLUGIN_SKIP_REGISTRATION OR NOT BUILD_SHARED_LIBS)
```

Comment:
let's add note that `NOT BUILD_SHARED_LIBS` is needed to registering in plugins.hpp